src/coreclr/tools/aot/ILCompiler.RyuJit/JitInterface/CorInfoImpl.RyuJit.cs

```diff
@@ -2203,5 +2203,59 @@ private int getExactClasses(CORINFO_CLASS_STRUCT_* baseType, int maxExactClasses
             Debug.Assert(index <= maxExactClasses);
             return index;
         }
+
+        private bool getReadonlyStaticFieldValue(CORINFO_FIELD_STRUCT_* fieldHandle, byte* buffer, int bufferSize)
+        {
+            FieldDesc field = HandleToObject(fieldHandle);
+            if (field.IsStatic && !field.IsThreadStatic && field.IsInitOnly && field.OwningType is MetadataType owningType)
+            {
+                PreinitializationManager preinitManager = _compilation.NodeFactory.PreinitializationManager;
+                if (preinitManager.IsPreinitialized(owningType))
+                {
+                    TypePreinit.ISerializableValue value = preinitManager
+                        .GetPreinitializationInfo(owningType).GetFieldValue(field);
+
+                    if (value == null)
+                    {
+                        *((nint*)buffer) = 0;
+                        return true;
+                    }
+
+                    object data;
+                    if (value.GetRawData(_compilation.NodeFactory, out data))
+                    {
+                        switch (data)
+                        {
+                            case byte[] bytes when bytes.Length <= bufferSize:
+                                bytes.AsSpan().CopyTo(new Span<byte>(buffer, bufferSize));
+                                return true;
+
+                            case FrozenStringNode str:
+                                *((nint*)buffer) = ObjectToHandle(str);
+                                return true;
+
+                            case FrozenObjectNode node:
+                                *((nint*)buffer) = ObjectToHandle(node);
+                                return true;
+                        }
+                    }
+                }
+            }
+            return false;
+        }
+
+        private CORINFO_CLASS_STRUCT_* getObjectType(void* objPtr)
+        {
+            object obj = HandleToObject((IntPtr)objPtr);
+            if (obj is FrozenStringNode)
+            {
+                return ObjectToHandle(_compilation.TypeSystemContext.GetWellKnownType(WellKnownType.String));
+            }
+            else if (obj is FrozenObjectNode)
+            {
+                return null; // TODO: how to implement properly?
```

Comment:
@MichalStrehovsky how do I implement it for `FrozenObjectNode` ? 🙂

Reply:
Grab the type from FrozenObjectNode._data.Type has a TypeDesc (you need to make a public API on FrozenObjectNode to get it).

Then just ObjectToHandle that type, like for the string above.

Reply:
Thanks, done!